v2.0/table-expressions.md

```diff
@@ -141,55 +207,101 @@ For example:
+~~~
+~~~
++---------+
+| column1 |
++---------+
+| a       |
+| b       |
+| c       |
++---------+
+~~~
+
+~~~sql
+> SELECT * FROM (VALUES ('a'), ('b'), ('c')) WITH ORDINALITY;
+~~~
+~~~
++---------+------------+
+| column1 | ordinality |
++---------+------------+
+| a       |          1 |
+| b       |          2 |
+| c       |          3 |
++---------+------------+
+~~~
+
+{{site.data.alerts.callout_info}}<code>WITH ORDINALITY</code> necessarily prevents some optimizations of the surrounding query. Use it sparingly if performance is a concern, and always check the output of <a href="explain.html">EXPLAIN</a> in case of doubt. {{site.data.alerts.end}}
+
 ## Join Expressions
 
+Join expressions combine the results of two or more table expressions
+based on conditions on the values of particular columns.
+
+See [Join Expressions](joins.html) for more details.
+
+## Using Other Queries as Table Expressions
+
+The following sections describe how to use the results produced by
+another SQL query or statement as a table expression.
+
+### Subqueries as Table Expressions
+
+Any [statement-like
+query](relational-expressions.html#statement-like-queries) enclosed
+between parentheses can be used as a table expression, including
+[simple `SELECT` clauses](select-clause.html). This is called a
+"[subquery](subqueries.html)".
+
 Syntax:
 
-~~~ shell
-# Inner joins:
-<table expr> [ INNER ] JOIN <table expr> ON <val expr>
-<table expr> [ INNER ] JOIN <table expr> USING(<colname>, <colname>, ...)
-<table expr> NATURAL [ INNER ] JOIN <table expr>
-<table expr> CROSS JOIN <table expr>
+~~~
+( ... subquery ... )
+~~~
 
-# Left outer joins:
-<table expr> LEFT [ OUTER ] JOIN <table expr> ON <val expr>
-<table expr> LEFT [ OUTER ] JOIN <table expr> USING(<colname>, <colname>, ...)
-<table expr> NATURAL LEFT [ OUTER ] JOIN <table expr>
+For example:
 
-# Right outer joins:
-<table expr> RIGHT [ OUTER ] JOIN <table expr> ON <val expr>
-<table expr> RIGHT [ OUTER ] JOIN <table expr> USING(<colname>, <colname>, ...)
-<table expr> NATURAL RIGHT [ OUTER ] JOIN <table expr>
+~~~sql
+> SELECT c+2                          FROM (SELECT COUNT(*) AS c FROM users);
+> SELECT *                            FROM (VALUES(1), (2), (3));
+> SELECT firstname || ' ' || lastname FROM (TABLE employees);
 ~~~
 
-These expressions designate the
-[SQL join operation](https://en.wikipedia.org/wiki/Join_(SQL)) on the
-two operand table expressions.
+{{site.data.alerts.callout_info}}See also <a href="subqueries.html">Subqueries</a> for more details and performance best practices.{{site.data.alerts.end}}
 
-Currently works only with small data sets; find more info in our [blog post](https://www.cockroachlabs.com/blog/cockroachdbs-first-join/).
+{{site.data.alerts.callout_info}}To use other statements that produce data in a table expression, for example `SHOW`, use the <a href="#using-the-output-of-other-statements">square bracket notation</a>.{{site.data.alerts.end}}
```

Comment:
I'd just include this in the previous callout, as a separate paragraph. 

Also, you can't use backticks inside a callout, so you need to use `<code>SHOW</code>`.